pkgs/development/libraries/mpich/default.nix

```diff
@@ -22,6 +22,8 @@ stdenv.mkDerivation  rec {
     "--enable-shared"
     "--enable-sharedlib"
     "--with-pm=${withPm}"
+    "FFLAGS=-fallow-argument-mismatch" # https://github.com/pmodels/mpich/issues/4300
```

Comment:
It is possible that this is already obsolete with the next version of mpich (which has just released in the past week).